### DISCUSSION TITLE
Commenting on a run_id after the experiment is finished

### DISCUSSION THREAD

**Discussion by Quantfoam** [microsoft/Qcodes#5745] [Q&A] [answered]
Hi all,

I am using Qcodes to acquire data and I am plotting it using Pyplotter (https://github.com/pyplotter/pyplotter). Sometimes I would like to add some metadata to a specific run_id with some comments about the run. I can easily extract the existing metadata in a jupyter notebook but it seems difficult to modify anything. I saw that in your QCoDeS experiments widget you have a notes section so I guess you thought about it. Is it possible to add note to an experiment afterwards?

Thanks

**Comment by edumur** [comment]
Our policy on the pyplotter is to *never* changed the database.
We then save our comment on an associated json file, the same file as to "star" or "hide" a run:
![aa](https://github.com/microsoft/Qcodes/assets/13218882/df5bc55b-c17e-4048-9be0-ba009356a9df)

**Reply by Quantfoam** [reply]
Thanks a lot! Actually wouldn't be better to add the json file to the database metadata? I am just thinking that if I open the same database with Pyplotter on a different computer I would have also to look for the runinfo.json file.

**Reply by edumur** [reply]
Hum I am not sure.
I am really shy about touching the db from qcodes...

In our lab the db are saved on a server and different computers access them.
The json being stored along side the db, everything works well.

**Reply by Quantfoam** [reply]
I see. Actually sharing a common json file is also doable in our case. Will try it out

**Comment by astafan8** [comment, marked as answer]
>  I saw that in your QCoDeS experiments widget you have a notes section so I guess you thought about it. Is it possible to add note to an experiment afterwards?

yes, in general, that's supported by qcodes through the `DataSet.add_metadata` method, so perhaps give it a try? We don't advertise this method a lot but it exists and works (also covered with tests) (and it has a few limitations), and this is exactly what the experient widget makes use of. https://microsoft.github.io/Qcodes/examples/DataSet/DataSet-class-walkthrough.html#Note-on-metadata
http://microsoft.github.io/Qcodes/api/dataset/index.html#qcodes.dataset.DataSetProtocol.add_metadata

**Reply by Quantfoam** [reply]
Thanks, I will check both options!
